lib/hsa/mcwamp_hsa.cpp

```diff
@@ -3383,16 +3368,13 @@ class HSAContext final : public KalmarContext
         STATUS_CHECK(status, __LINE__);
 
         // Iterate over the agents to find out gpu device
-        status = hsa_iterate_agents(&HSAContext::find_gpu, &agents);
-        STATUS_CHECK(status, __LINE__);
-
-        // Iterate over agents to find out the first cpu device as host
-        status = hsa_iterate_agents(&HSAContext::find_host, &host);
+        std::vector<hsa_agent_t>* agent_vectors[] = {&agents, &cpu_agents};
+        status = hsa_iterate_agents(&HSAContext::find_agents, &agent_vectors);
         STATUS_CHECK(status, __LINE__);
 
         for (int i = 0; i < agents.size(); ++i) {
             hsa_agent_t agent = agents[i];
-            auto Dev = new HSADevice(agent, host, i);
+            auto Dev = new HSADevice(agent, cpu_agents[i%cpu_agents.size()], i);
```

Comment:
@scchan this doesn't seem to be correct, is it possible to understand the affinity of a GPU agent to a CPU?

Reply:
I went through the HSA headers and it doesn't seem to provide any affinity info?
There's a node # but on my system the CPUs and GPU are all on different nodes